### ISSUE TITLE
Time parsing not lenient like moment

### ISSUE THREAD

**Issue by jacobg** [iamkun/dayjs#2748]
**Describe the bug**
Moment.js is able to do lenient time parsing by using parse format 'HH:mm a'. With that format, you can parse '4:26 PM', '4:26 pm', '4:26pm', '16:26', etc. But in dayjs, the format is exact and anything else is an `Invalid Date`. Is there a way to do more lenient time parsing?

**Expected behavior**
Emulate moment lenient time parsing behavior.

**Information**
 - Day.js Version [e.g. v1.0.0]  1.11.13
 - OS: [e.g. iOS] Mac OS 14.7
 - Browser [e.g. chrome 62] Chrome 129
 - Time zone: [e.g. GMT-07:00 DST (Pacific Daylight Time)]  GMT-4 EDT

**Comment by deepanshu188** [comment]
![20241124_13h29m58s_grim](https://github.com/user-attachments/assets/47a8b18f-0537-4e8f-a372-5b0397185dbb)

You mean this?

**Comment by jacobg** [comment]
@deepanshu188 No, your screenshot is formatting/displaying. I'm talking about parsing, e.g.,
```js
dayjs('4:26 PM', 'HH:mm a')
```
